### ISSUE TITLE
PSUContol - Realy always ON (activated)

### ISSUE THREAD

**Issue by DanRom624** [kantlivelong/OctoPrint-PSUControl#118]
#### What were you doing?

1. I am trying to use a TONGLING 5VDC relay (specs JQC-3FF-S-Z)
    **No matter what is the PSU Control status, the relay is always "activated"**

2. I have connected the Pi to the Relay as follows:
       GPIO Pin 4 (5V) --> VCC pin on relay board
       GPIO Pin 6 (ground) --> GND pin on relay board
       GPIO Pin 7 (4) --> IN pin on relay board

3. I have set up the PSUControl as follows:
       Show warning dialog...         ->  UNCHECKED
       GPIO Mode                            ->  BOARD
       Switching Method                 ->  GPIO Pin
       On/Off GPIO Pin                    ->  7
       Invert                                      ->  UNCHECKED
       Enable switching with G-Code ->   UNCHECKED
       Sensing Method                    ->  Internal
       Automatically turn PSU ON  ->  UNCHECKED
       Post On Delay                        ->  0 (zero) 
       Automatically turn PSU OFF ->  UNCHECKED
       Disconnect on power off      ->  UNCHECKED

4. As a control, I measured tension between GPIO Pin 6 (ground) and GPIO Pin 7 (4) and found 3.28V as expected (not exactly 3.30V, but fair enough...) when PSU Control is ON, 0.00V when it is OFF - good!

5. For testing purposes, I connected a light bulb (instead of the printer) to the "mains side" of the Relay. One wire goes directly to the bulb, the other one is connected to the COM on the Relay. One additional wire connects the bulb to the NO (Normally Open) connector on the Relay   

#### What did you expect to happen?
When pressing the "lightning icon" on PSU Control, it shall turn Green, and the Relay should switch ON (click from idle to activated) thus closing the circuit on the mains side of the relay, and, ultimately, switching on the bulb (printer). However...

#### What happened instead?
... as soon as I connect my Pi to the Relay (see no.2 above), the Relay clicks to activated, REGARDLESS if the PSU Control is ON or OFF (consequently, the bulb stays always on).  If I disconnect the GPIO Pin 7, though, the Relay clicks back to its idle position (and the bulb goes off)

#### Version of OctoPrint-PSUControl
0.1.7

#### Operating System running OctoPrint
MacOS Mojave (OS X version 10.14.1)
OctoPrint version 1.3.10

#### Printer model & used firmware incl. version
Anet A8 - Marlin firmware ver. 1.1.9 (this is not really important, as the light bulb test failed before I ever tried to control my printer via PSU Control)

#### Many thanks !

**Comment by kantlivelong** [comment]
Hi,

This sounds like a wiring issue.

Quick Google search is suggesting that the IN pin on the relay turns on when LOW and off when HIGH. 

If you touch the IN pin to ground does the relay turn on?

**Comment by DanRom624** [comment]
Yes !
If you mean: disconnect IN from GPIO 7, and touch it to GND... then yes, it switches on and off

**Comment by kantlivelong** [comment]
Right so then it's a wiring issue. 

See: https://raspberrypi.stackexchange.com/questions/27928/power-a-5v-relay-from-gpio-pins

**Comment by DanRom624** [comment]
This is exactly the Relay module I am using:

![image](https://user-images.githubusercontent.com/18577807/50715115-f5100380-107b-11e9-8e95-a882330c1477.png)

**Comment by DanRom624** [comment]
This is exactly the relay module I am using:
![image](https://user-images.githubusercontent.com/18577807/50715171-45876100-107c-11e9-8a18-2001824a8f04.png)

**Comment by kantlivelong** [comment]
Right. The URL I provided above should get you going.

**Comment by nionio6915** [comment]
1. The Tongling moniker does not mean anything in this context. The important thing is that you have a 5V relay module.  

2. This page might be a little more beginner friendly-
https://www.instructables.com/id/5V-Relay-Raspberry-Pi/

3. PSU control - This is how I have my 5V relay module configured.  Mine is different brand/type, but I am using pins 4-6-8 for 5V-Gnd-GPIO.

![image](https://user-images.githubusercontent.com/17693150/50717200-33d99580-104b-11e9-8255-99bb6f9af254.png)

**Comment by DanRom624** [comment]
Thank you both Bruce (kantlivelong) and Dennis (nionio6915) for your precious insight.

@nionio6915 
My relay module is indeed a "Low Level Trigger" (LLT)
I was looking for it's specs, but this specific feature was already stated under my eyes: too much close to be seen ;-)
(I found today that the trigger mode is usually written on the flip side of each module :-))

![ges544bos22fz aj1ghosa_thumb_c878](https://user-images.githubusercontent.com/18577807/50729443-e12ad700-1139-11e9-9a53-344c9e383894.jpg)

The key issue here, is exactly that: how is the module triggered? Is it a Low or a High Level Trigger? Most of the examples that can be found on the net (as the one that you suggested, for instance) are based on a High Level Trigger (HLT) module. 
Converting a LLT into a HLT module is quite simple, though. This brings me to what Bruce suggested yesterday...

@kantlivelong  

> Right so then it's a wiring issue.
> 
> See: https://raspberrypi.stackexchange.com/questions/27928/power-a-5v-relay-from-gpio-pins

Yes, that solution works.
However, it is not ver intuitive for a beginner.
Besides, it is designed for a relay (as such) not for a ready made "relay module" (as mine)
I found this alternative solution, which also works, and is very intuitive
https://www.youtube.com/watch?v=-wiygjMviFo
It also shows how it is practically achieved, which is very useful for beginners.

Again, thank you both, and thank you Aviprink for his tutorial
